src/librawspeed/io/Buffer.h

```diff
@@ -22,15 +22,19 @@
 #pragma once
 
 #include "common/Common.h" // for uchar8, uint32, uint64
+#include "io/IOException.h"// for ThrowIOE
 #include "io/Endianness.h" // for getByteSwapped
 #include <algorithm>       // for swap
 
 namespace RawSpeed {
 
-// All file maps have this much space extra at the end. This is useful for
-// BitPump* that needs to have a bit of extra space at the end to be able to
-// do more efficient reads without crashing
-#define FILEMAP_MARGIN 16UL
+// This allows to specify the nuber of bytes that each Buffer needs to
+// allocate additionally to be able to remove one runtime bounds check
+// in BitSream::fill. There are two options:
+// 0 : allocate exactly as much data as required
+// 4 : add minimum number of bytes to keep all BitStream implementations happy
+//     this disables the bounds checking, saves about 1% on modern CPUs
+#define BUFFER_PADDING 0
```

Comment:
This really is a non-issue: The user has to use the library in way very likely no one uses it right now (allocating the memory themself instead of using `FileReader`) _and_ has to modify the source code of the library to create a potential problem. And if he modifies the code he can introduce all kinds of issues if does not know what he is doing. This define is about as safe as some comment that explains how the code could be changed.